pallets/transfer-allowlist/src/lib.rs

```diff
@@ -0,0 +1,565 @@
+				Some(delay) => AllowanceDetails {
+					allowed_at: <frame_system::Pallet<T>>::block_number().saturating_add(delay),
+					..AllowanceDetails::default()
+				},
+				_ => AllowanceDetails::default(),
+			};
+			if !<AccountCurrencyTransferAllowance<T>>::contains_key((
+				&account_id,
+				&currency_id,
+				&receiver,
+			)) {
+				Self::increment_or_create_allowance_count(&account_id, &currency_id)?;
+				T::ReserveCurrency::reserve(&account_id, T::Deposit::get())?;
+			};
+			<AccountCurrencyTransferAllowance<T>>::insert(
+				(&account_id, &currency_id, &receiver),
+				&allowance_details,
+			);
+
+			Self::deposit_event(Event::TransferAllowanceCreated {
+				sender_account_id: account_id,
+				currency_id,
+				receiver,
+				allowed_at: allowance_details.allowed_at,
+				blocked_at: allowance_details.blocked_at,
+			});
+			Ok(())
+		}
+
+		#[transactional]
+		#[pallet::call_index(1)]
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(2, 2).ref_time())]
+		/// Restricts a transfer allowance for a sending account/currency/receiver location to:
+		/// - either the current block + delay if a delay is set
+		/// - or the current block if no delay is set
+		pub fn remove_transfer_allowance(
+			origin: OriginFor<T>,
+			currency_id: CurrencyIdOf<T>,
+			receiver: Location,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+
+			let blocked_at = match Self::sender_currency_delay(&account_id, currency_id) {
+				Some(delay) => <frame_system::Pallet<T>>::block_number().saturating_add(delay),
+				_ => <frame_system::Pallet<T>>::block_number(),
+			};
+			match <AccountCurrencyTransferAllowance<T>>::get((&account_id, &currency_id, &receiver))
+			{
+				Some(existing_allowance) => {
+					let allowance_details = AllowanceDetails {
+						blocked_at,
+						..existing_allowance
+					};
+					<AccountCurrencyTransferAllowance<T>>::insert(
+						(&account_id, &currency_id, &receiver),
+						&allowance_details,
+					);
+					Self::deposit_event(Event::TransferAllowanceRemoved {
+						sender_account_id: account_id,
+						currency_id,
+						receiver,
+						allowed_at: allowance_details.allowed_at,
+						blocked_at: allowance_details.blocked_at,
+					});
+					Ok(())
+				}
+				None => Err(DispatchError::from(Error::<T>::NoMatchingAllowance)),
+			}
+		}
+
+		#[transactional]
+		#[pallet::call_index(2)]
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1, 2).ref_time())]
+		/// Removes a transfer allowance for a sending account/currency and receiving location
+		/// Decrements or removes the sending account/currency count.
+		pub fn purge_transfer_allowance(
+			origin: OriginFor<T>,
+			currency_id: CurrencyIdOf<T>,
+			receiver: Location,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+			match <AccountCurrencyTransferAllowance<T>>::get((&account_id, &currency_id, &receiver))
+			{
+				Some(_) => {
+					T::ReserveCurrency::unreserve(&account_id, T::Deposit::get());
```

Comment:
@mustermeiszer @mikiquantum should we use `pallet-fees` here?

Reply:
Pallet fees do not support reserve/unreserve, afaik, only one way transfer/burn/etc

Reply:
But should `T::Deposit::get()` come from a key? At least we do so for `Anchors::pre_commit()`.

Maybe `reserve/unreserve` actions need to be added to `pallet-fees`. 🤔

Reply:
Ah yes @lemunozm I would definitely define a key in the pallet fees and read the amount from it, that even if for now we do not support the reserve/unreserve logic yet there. That way the UIs can know up front operational costs.

Reply:
👍  Pallet is switched over to use fee key & getting fee val with fees pallet.  Also have mock set up to set key val for allowance creation with fees pallet to ensure fees pulled w/ correct key -- `FeeKey` used in pallet itself is still loosely coupled in allowance pallet.

Reply:
I left a comment regarding the mock part @thea-leake.

Regarding `reserve()`/`unreserve()`, @mikiquantum I've created an issue to add them to `pallet-fees`: #1282